packages/iov-keycontrol/src/keyring-entries/ed25519.ts

```diff
@@ -177,4 +181,23 @@ export class Ed25519KeyringEntry implements KeyringEntry {
     }
     return privkey;
   }
+
+  private buildLocalIdentity(bytes: PublicKeyBytes, label: string | undefined): LocalIdentity {
+    const pubkey: PublicKeyBundle = {
+      algo: Algorithm.ED25519,
+      data: bytes,
+    };
+    return {
+      pubkey,
+      label,
+      id: Ed25519KeyringEntry.identityId({ pubkey }),
+    };
+  }
+
+  private randomId(): string {
+    // this can be pseudo-random, just used for internal book-keeping
+    const index = Math.random() * 1000000000;
```

Comment:
this doesn't seem safe since `Math` seems like it would have a global seed so could easily be reseeded to accidentally create surprising keyring collisions e.g. in scientific simulations that reseed themselves. i would suggest we avoid the global random context and instead have a private random context that cannot be externally reseeded accidentally.

Reply:
can use `new` form without seed parameter to get automatic seeding https://github.com/davidbau/seedrandom